test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1398,8 +1398,7 @@ private void randomlyResetClients() {
         }
     }
 
-    private void wipePendingDataDirectories() {
-        assert Thread.holdsLock(this);
+    public synchronized void wipePendingDataDirectories() {
```

Comment:
I can't see any _obvious_ problems that arise from exposing this, but the API is kinda complicated so I'd like a second opinion from @ywelsch.